var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -361,6 +367,9 @@ def cmake_args(self):
                 [define("TPL_ENABLE_METIS", False), define("TPL_ENABLE_ParMETIS", False)]
             )
 
+        if "+pamgen" in self.spec:
```

Comment:
I would prefer a slightly different syntax (suggested by Spack developers on previous PRs):
```
options.extend(from_variant("Seacas_ENABLE_Pamgen", "pamgen"))
options.extend(from_variant("TPL_ENABLE_Pamgen", "pamgen"))
```
Not sure if both are needed -- does the `Seacas_ENABLE_Pamgen` get set if `TPL_ENABLE_Pamgen` is specified?

Also not sure whether we need a `Pamgen_ROOT` so CMake can find Pamgen, or is that already taken care of elsewhere...

Reply:
It does seem to find pamgen when it builds Trilinos with +pamgen as the configure errors out if it is not built properly. I think the Seacas_ENABLE_Pamgen is unneeded. Changing and testing currently.

Reply:
> It does seem to find pamgen when it builds Trilinos with +pamgen as the configure errors out if it is not built properly. I think the Seacas_ENABLE_Pamgen is unneeded. Changing and testing currently.

It looks like there is an incorrect test in the SEACAS CMakelists.txt files in IOSS -- They are testing for `Seacas_ENABLE_Pamgen` instead of `TPL_ENABLE_Pamgen` which is the form used for other TPL's.  I've fixed this in a commit today, but that won't work for previous releases...   Probably need to leave the `Seacas_ENABLE_Pamgen` define in for now and we can tie it to previous versions on the next release...

Reply:
For now, I will leave both in but make the refactor then. I did refactor it and it is working still.